frontend/src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,56 @@
+import { BiCaretDown, BiCaretUp } from 'react-icons/bi'
+import {
+  forwardRef,
+  Icon,
+  InputRightElement,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput as ChakraNumberInput,
+  NumberInputField,
+  NumberInputProps as ChakraNumberInputProps,
+  NumberInputStepper,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+import { omit } from '@chakra-ui/utils'
+
+import { BxsCheckCircle } from '~assets/icons/BxsCheckCircle'
+
+export interface NumberInputProps extends ChakraNumberInputProps {
+  /**
+   * Whether the input is in a prefilled state.
+   */
+  isPrefilled?: boolean
+  /**
+   * Whether the input is in a success state.
+   */
+  isSuccess?: boolean
+}
+
+export const NumberInput = forwardRef<NumberInputProps, 'input'>(
+  (props, ref) => {
+    const { isSuccess } = props
+    const inputStyles = useMultiStyleConfig('NumberInput', props)
+    // Omit extra props so they will not be passed into the DOM and trigger
+    // React warnings.
+    const inputProps = omit(props, ['isSuccess', 'isPrefilled'])
+
+    return (
+      <ChakraNumberInput {...inputProps} ref={ref}>
+        <NumberInputField sx={inputStyles.field} />
+        {isSuccess && (
+          <InputRightElement sx={inputStyles.success}>
+            <Icon as={BxsCheckCircle} />
+          </InputRightElement>
+        )}
+        <NumberInputStepper>
+          <NumberIncrementStepper>
+            <Icon as={BiCaretUp} transform="translate(0, 3px)" />
```

Comment:
would it be better to convert the px to rem?

Reply:
not for adhoc changes like this, imagine 3px is 0.1875rem :P

Reply:
This is moot anyways since the design is changing lul